### PR TITLE
I393 data ingestion commands: add option -F for continue mini file from

### PR DESCRIPTION
**Pull request by damhuonglan**
### What does this PR do?
Issue #393 states that segdtoph5 doesn't continue mini file (miniPH5_xxxxx.ph5) index from option `-S first_mini`. However, that option is to state the first mini file index of all. In segdtoph5 there is no option to continue mini file index from. 

This PR adds option `-F from_mini `to continue index for mini file from, in which the mini files created will not be greater than 100GB. This option will not be associated with option `-M num_mini`.

The help in segdtoph5 is modified and added example as following to avoid misleading: 
```

  -M num_mini, --num_mini=num_mini
                        Create a given number of miniPH5 files. Ex: -M 38
  -S first_mini, --first_mini=first_mini
                        The index of the first miniPH5_xxxxx.ph5 file of all.
                        Ex: -S 5
  -F from_mini, --from_mini=from_mini
                        The index to continue miniPH5_xxxxx.ph5 file from. Do
                        not associate with option -M. Ex: -F 25
```

This option are added to mstoph5, 130toph5, 125toph5, seg2toph5 too.

### Checklist
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] All tests pass.
- [ ] Any new or changed features have are documented.
- [ ] Changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

File: ph5/utilities/segd2ph5.py [hunk 1]
```diff
@@ -177,13 +177,27 @@ def get_args():
...
-                       help="The index of the first miniPH5_xxxxx.ph5 file.",
+                       help=("The index of the first miniPH5_xxxxx.ph5 file "
+                             "of all. Ex: -S 5"),
                        metavar="first_mini", type='int', default=1)
 
+    oparser.add_option("-F", "--from_mini", dest="from_mini",
+                       help=("The index to continue miniPH5_xxxxx.ph5 file "
+                             "from. Do not associate with option -M. "
+                             "Ex: -F 25"),
+                       metavar="from_mini", type='int', default=None)
```
Comment (dsentinel) [inline ph5/utilities/segd2ph5.py:193]: metavar isn't necessary for named arguments if setting to the same as the arg name

File: ph5/utilities/segd2ph5.py [hunk 2]
```diff
@@ -227,6 +243,14 @@ def get_args():
     else:
         raise Exception("No outfile (PH5) given.\n")
 
+    # from_mini and num_mini must not coexist
+    if (NUM_MINI is not None) and (FROM_MINI is not None):
+        raise Exception('Option -M and option -F must not be '
+                        'used at the same time.')
+
+    if ONE_PER_MINI and (FROM_MINI is None):
+        raise Exception('Option -O must be associated with option -F.')
```
Comment (dsentinel) [inline ph5/utilities/segd2ph5.py:252]: Mutual exclusion can be handled by argparse
https://docs.python.org/3/library/argparse.html#mutual-exclusion

File: ph5/utilities/segd2ph5.py [hunk 3]
```diff
@@ -1030,7 +1083,13 @@ def get_node(sd):
...
             rows, keys = EX.ph5_g_maps.read_index()
             INDEX_T_MAP = Rows_Keys(rows, keys)
-
+            if FROM_MINI is not None:
+                highestMini = getHighestMini(INDEX_T_DAS)
+                if FROM_MINI < highestMini:
+                    LOGGER.error("FROM_MINI must be greater than %s, "
+                                 "the highest mini file in ph5." % highestMini)
+                    EX.ph5close()
+                    return 1
```
Comment (dsentinel) [inline ph5/utilities/segd2ph5.py:1092]: what does return 1 mean?

**Comment by damhuonglan** [comment]
List of checks need to be done:

- [ ] Argument helps and examples make sense

- [ ] -M associate with -F gives error

- [ ] -O without -F gives error

- [ ] -M still works the same as before

- [ ] -F with value less than the highest mini file gives error

- [ ] -F without -O: start mini file from F's value, 1 file can have more than one das's data

- [ ] -F with -O: start mini file from F's value, 1 file can have only one das's data

You can add any more checks that you feel need too.

**Review by hrotman-pic — COMMENTED** [review]
It looks like the -F option permits adding same DAS to new minis, making it easier to add service runs to an existing archive without having to resubmit the entire archive.  Please see Working/Holly_append_test/PR426/test1.  I have not validated that test archive since that is out of scope.

Overall the help is more informative and makes more sense.  One small suggestion for -M flag, I think it would be good if help said also 'Recommend using when creating a new PH5' since I think that is the intended use of -M.

**Comment by hrotman-pic** [comment]
To clarify, I have *only* tested segdtoph5 as that was the original scope of this PR and I had a set of comparison test archives.  I recommend additional reviewers for the other ingestion commands.

**Comment by ascire-pic** [comment]
@hrotman-pic is going to test mstoph5 and 125atoph5
@akram-pic is going to test seg2toph5 and 130toph5

**Comment by damhuonglan** [comment]
> To clarify, I have _only_ tested segdtoph5 as that was the original scope of this PR and I had a set of comparison test archives. I recommend additional reviewers for the other ingestion commands.

@hrotman-pic Thanks for your review. I already put your recommended help message to -M. I already added -M to other ingestion commands too. 

> @hrotman-pic is going to test mstoph5 and 125atoph5
> @akram-pic is going to test seg2toph5 and 130toph5

All data ingestion commands can be tested in this PR (checkout branch: i393_segdtoph5_optionS) except for mstoph5 is in PR #429 (checkout branch: i393-mstoph5_optionS)

**Review by hrotman-pic — CHANGES_REQUESTED** [review]
Based on the meeting, this review is requesting:
Removal of -F option for all ingestion commands changed under this PR, so that users cannot recreate the problem of data from one DAS in two or more mini files.

The changes to the help for other options, such as 'Ex: -M 38' are still useful in my view.

I am not sure about keeping 'Recommend using when creating a new PH5' for the -M option help (for all ingestion commands in this PR).  If we keep it, I think it will help if the help for the -M option also says that adding new data to an existing PH5 may not create new mini files.

Going slightly out of the scope of the discussion: I think it would help standalone users to have an example of an ingestion command listed in the help, such as: 
segdtoph5 -n master.ph5 -c 6 -U 18N -M 20
And to state that -S defaults to 1 if not specified in the help for -S option.

**Comment by damhuonglan** [comment]
replaced with PR#434

**Pull request closed without merging**